dev/tests/integration/framework/Magento/TestFramework/Annotation/DbIsolation.php

```diff
@@ -17,6 +20,85 @@ class DbIsolation
      */
     protected $_isIsolationActive = false;
 
+    /**
+     * @var string[]
+     */
+    private $dbStateTables = [
+        'catalog_product_entity' => 'assertIsEmpty',
+        'eav_attribute' => 'eavAttributeAssert',
+        'catalog_category_entity' => 'assertTwoRecords',
+        'eav_attribute_set' => 'attributeSetAssert',
+        'store' => 'assertTwoRecords'
+    ];
+
+    /**
+     * @param array $data
+     * @throws \Exception
+     */
+    private function attributeSetAssert(array $data)
+    {
+        if (count($data) > 9) {
+            return array_slice($data, 9, count($data) - 9);
+        }
+
+        return false;
+    }
+
+    /**
+     * @param array $data
+     * @throws \Exception
+     */
+    private function assertTwoRecords(array $data)
+    {
+        //2 default records
+        if (count($data) > 2) {
+            return array_slice($data, 2, count($data) - 2);
+        }
+
+        return false;
+    }
+
+    /**
+     * @param array $data
+     * @throws \Exception
+     */
+    private function eavAttributeAssert(array $data)
+    {
+        //178 - default number of attributes
+        if (count($data) > 178) {
```

Comment:
For CE/EE/B2B/Whatever edition it may give different results

Reply:
Of course. Need to override files for each version or make it smart

Reply:
we can do it for first iteration, but better to have more smart logic and cache "empty" state before tests run (we can introduce some IsolationTestState registry and compare each test run with date from this storage